### DISCUSSION TITLE
useMutation for request that doesn't do CRUD?

### DISCUSSION THREAD

**Discussion by hornetDC** [TanStack/query#4947] [Q&A] [answered]
Hi, I am doing a PR on my project and I came across a component where form submit is calling `refetch()` from disabled `useQuery`.
It feels wrong but form submit doesn't update anything, it's a POST request that sends form data and receives formatted object in response, so it technically can be called "fetching".

**Comment by louis-young** [comment, marked as answer]
Hi 👋 

It sounds like you might be describing a GET request (a request that requests some data, given some parameters) that's using the POST verb. POST requests typically result in a change on the server; if this isn't the case, you might want to consider changing the HTTP verb.

Does this help? 🙂

**Reply by hornetDC** [reply]
It could be a case of incorrect verb, I haven't looked at BE code. Still I am not sure if `useQuery` `refetch()` or `useMutation` `mutate()` is appropriate for such form submit.

**Reply by louis-young** [reply]
A disabled query re-fetch is valid but somewhat unusual. You could likely do this with a dependant query without calling `refetch` manually. If it's requesting data from the server without resulting in change then it should likely be a GET request. If it is a GET request, it should be a query and not a mutation.
